GDJS/Runtime/runtimescene.ts

```diff
@@ -744,6 +748,14 @@ namespace gdjs {
             } else {
               this._allInstancesList.push(list[j]);
             }
+            const layerName = list[j].getLayer();
+            const zOrder = list[j].getZOrder();
+            if (
+              !this.hasLayer(layerName) ||
+              this.getLayer(layerName).getHighestZOrder() < zOrder
```

Comment:
We can be confident that `getLayer` gives the right layer.

Reply:
You're right, one check fewer!